cabra/src/pages/components/CheckBox.tsx

```diff
@@ -0,0 +1,18 @@
+import { ITask } from "../../types/task";
+import tw from "twin.macro";
+import { updateTask } from "../../api/tasks";
+
+const CheckBoxStyle = tw`ml-auto flex items-center gap-2 p-2 rounded checked:bg-blue-500 checked:border-blue-500`;
+
+const changeFlagIsDone = (task: ITask) => {
+    task.isDone = ! task.isDone;
+    updateTask(task.id, task);
+};
+
+const CheckBox = (task: ITask) => {
+    return (
+        <input type="checkbox" css={CheckBoxStyle} checked={task.isDone} onClick={() => changeFlagIsDone(task)}/>
+    );
+};
+
+export default CheckBox;
```

Comment:
A couple of things about this component:
- first of all, good job with making a separate component out of it! That nicely helps with the encapsulation of styles.
- next thing is `props`. In React you usually pass data to components through `props`.`Props` are received as an object in the first (and almost always only) argument of functional components like the one above. Using props allows you to use the components with nice JSX syntax e.g `<CheckBox task={task}/>` which is de facto the standard way to use a component. You can see how props are used in [Task component](https://github.com/it-goats/it-goats/blob/b0ece95c24542d255aef416f289d8221d93e8902/cabra/src/pages/components/Task.tsx). It also a good practice to limit the number of props that a component receives as much as possible. The checkbox actually needs two things: its state to be passed to the `checked` prop and a handler that reacts to changes of said state. The logic connected to these attributes should be handled by the parent component that uses the checkbox.
- talking to the API: the `updateTask` function that you used, calls are backend API but doesn't really do anything else. One of the things it's not doing is refreshing other API calls responsible for getting task data from the backend. For that behaviour you need  to `useMutation` from `react-query`. `react-query` is a async state managing library that is used to store the results of API calls and properly sync our frontend data with any changes that might be happening to it. Updating a task should trigger a refetch of API calls that might now have stale data. To do that you need to invalidate queries that can be affected. You can see an example of how it's  done in [TaskEditPage](https://github.com/it-goats/it-goats/blob/b0ece95c24542d255aef416f289d8221d93e8902/cabra/src/pages/TaskEdit.tsx) or in [CreateTaskForm](https://github.com/it-goats/it-goats/blob/b0ece95c24542d255aef416f289d8221d93e8902/cabra/src/pages/components/CreateTaskForm.tsx) components.
- last thing is SRP: this checkbox shouldn't really be responsible for making any specific API calls or business logic, that responsibility should fall on the parent components that uses it while the checkbox stays dumb and lean. Also it's usually preferred to not mutate objects used inside React, because it can lead to unexpected behavior and makes the code harder to reason about, so instead of writing:
```js
task.isDone  = !task.isDone
updateTask(task.id, task);
```
you should usually write:
```js
const updatedTask = {...task, isDone: !task.isDone};
updateTask(updatedTask.id, updatedTask);
```
This way the original task object that might be used by many other components stays untouched.

Summing up what I'd like to see:
- taking the task updating logic outside of the checkbox, probably to the  `Task` component that uses it. And also using `react-query` to make sure the frontend state is nicely synced.
- refactor of the checkbox so that it uses props, see suggestion below

```suggestion
import tw, { styled } from "twin.macro";

import { InputHTMLAttributes } from "react";

const StyledCheckboxInput = styled.input(
  tw`ml-auto flex items-center gap-2 p-2 rounded checked:bg-blue-500 checked:border-blue-500`
);

export default function Checkbox(
  props: Omit<InputHTMLAttributes<HTMLInputElement>, "type">
) {
  return <StyledCheckboxInput {...props} type="checkbox" />;
}

```

PS: For handling checkbox changes, you should use `onChange` instead of `onClick` handler and use the `checked` attribute of the underlying HTML element:
```jsx
<Checkbox
  checked={false}
  onChange={(event) => {
    // do something with
   event.target.checked;
  }}
/>
```

Reply:
Chciałbym dopytać czy mam dobry pomysł, przy wywoływaniu funkcji z useMutation powinienem użyć konstrukcji async/await, złapać ewentualny wyjątek i wypisać komunikat w tasku?

Reply:
Jak najbardziej dobry, warto poinformować użytkownika w razie problemów z odhaczeniem zadania